packages/cmf/src/cmfConnect.js

```diff
@@ -272,14 +311,63 @@ export default function cmfConnect({
 				}
 			}
 
+			onEvent() {
+				return Object.keys(this.props).reduce(this.onEventDispatch, { toOmit: [] });
+			}
+
+			onEventDispatch(props, key) {
+				if (key.startsWith('on')) {
+					if (key.endsWith('Dispatch')) {
+						props.toOmit.push(key);
+						// eslint-disable-next-line no-param-reassign
+						props[key.replace('Dispatch', '')] = (event, data) => {
+							const payload = Object.assign(
+								{
+									event: serializeEvent(event),
+									data,
+								},
+								this.props[key],
+							);
+							this.props.dispatch(payload);
+						};
+					}
+					if (key.endsWith('ActionCreator')) {
+						props.toOmit.push(key);
+						const newKey = key.replace('ActionCreator', '');
+						let actionCreator = this.props[key];
+						let args;
+						if (typeof this.props[key] === 'object') {
+							actionCreator = this.props[key].id;
+							args = this.props[key].args;
+						}
+						// eslint-disable-next-line no-param-reassign
+						props[newKey] = (event, data) => {
+							if (!args) {
+								args = [
+									serializeEvent(event),
+									{
+										props: this.props,
+										data,
+									},
+								];
+							}
+							this.dispatchActionCreator(actionCreator, ...args);
+						};
+					}
+				}
+				return props;
+			}
+
 			dispatchActionCreator(actionCreatorId, event, data, context) {
 				const extendedContext = Object.assign({}, this.context, context);
 				this.props.dispatchActionCreator(actionCreatorId, event, data, extendedContext);
 			}
 
 			render() {
+				const { toOmit, ...handlers } = this.onEvent();
```

Comment:
Why is that in render instead of `getDispatchToProps` ?
Other props manipulation are in `getDispatchToProps` or `getStateToProps`

Reply:
this built in render because you need the router in the context.
This is an option I would like to do as optimisation in an other PR:
synchronise all router info into the store so we can remove the container of cmfConnect. This will increase performance I'm sure.
the issue:
mapStateToProps -> state
mapDispatch -> dispatch
merge -> only props.
So I can't have both.
The router is not accessible at this level at the moment so we can't do better without lots of change, but this is should be  added to the roadmap.